registry/handlers/layerupload.go

```diff
@@ -291,10 +329,15 @@ func (luh *layerUploadHandler) layerUploadResponse(w http.ResponseWriter, r *htt
 		return err
 	}
 
+	endRange := offset
+	if endRange > 0 {
+		endRange = endRange - 1
```

Comment:
Yes it was, my understanding is offset was returning the beginning of next range which is also the length.  However the returned range should be inclusive, excluding the first element of the next range. This fixed a bug in my code which was checking the length from the range by doing (end - start + 1).